src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/MSBuildProjectLoadListener.cs

```diff
@@ -10,41 +10,39 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor.Language;
-using Microsoft.AspNetCore.Razor.LanguageServer.Serialization;
 using Microsoft.AspNetCore.Razor.OmniSharpPlugin;
 using Microsoft.Build.Execution;
-using Microsoft.CodeAnalysis.Razor.ProjectSystem;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
 using OmniSharp;
 using OmniSharp.MSBuild.Notification;
 
 namespace Microsoft.AspNetCore.Razor.OmnisharpPlugin
 {
+    [Shared]
     [Export(typeof(IMSBuildEventSink))]
-    internal class ProjectLoadListener : IMSBuildEventSink
+    [Export(typeof(OmniSharpProjectSnapshotManagerChangeTrigger))]
+    internal class MSBuildProjectLoadListener : IMSBuildEventSink, OmniSharpProjectSnapshotManagerChangeTrigger
```

Comment:
`MSBuildProjectManager` and `IOmniSharpProjectSnapshotChangeTrigger`

Reply:
`MsBuildLoadManagerProject` and `UOmniSharpChangeProjectTriggerManager`

Reply:
Also can we please add `ICanHasRazorProjectChangeHandlerManagerResolverProviderFactoryFactory`?

Reply:
Obvs, consider it done.